source/Calamari/Deployment/PackageRetention/Model/Journal.cs

```diff
@@ -123,9 +150,28 @@ public int GetUsageCount(PackageIdentity package)
             return GetUsage(package).Count();
         }
 
-        public void ExpireStaleLocks()
+        public void ExpireStaleLocks(TimeSpan timeBeforeExpiration)
         {
-            throw new NotImplementedException();
+            try
+            {
+                using (var repository = repositoryFactory.CreateJournalRepository())
+                {
+                    foreach (var entry in repository.GetAllJournalEntries())
+                    {
+                        var usages = entry.GetLockDetails();
+                        var staleUsages = usages.Where(u => u.DateTime.Add(timeBeforeExpiration) <= DateTime.Now);
+
+                        foreach (var staleUsage in staleUsages)
+                        {
+                            entry.RemoveLock(staleUsage.DeploymentTaskId);
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                log.Error($"Unable to expire stale lock.{Environment.NewLine}{ex.ToString()}");
```

Comment:
stale lock => stale lock**s**

Reply:
Not sure what I was thinking here 😅. I've changed it to `stale package locks`.